src/ir/possible-contents.cpp

```diff
@@ -2135,6 +2135,9 @@ struct Flower {
   // For a non-full cone, we also reduce the depth as much as possible, so it is
   // equal to the maximum depth of an existing subtype.
   Index getNormalizedConeDepth(Type type, Index depth) {
+    if (type.isExact()) {
+      return 0;
+    }
```

Comment:
I think this might be a better place:
```diff
diff --git a/src/ir/possible-contents.h b/src/ir/possible-contents.h
index 538b23d4d..ab5b53436 100644
--- a/src/ir/possible-contents.h
+++ b/src/ir/possible-contents.h
@@ -164,8 +164,8 @@ public:
     assert(type != Type::none);
 
     if (type.isRef()) {
-      // For a reference, subtyping matters.
-      return fullConeType(type);
+      // For a reference, subtyping matters, unless it is exact.
+      return type.isExact() ? exactType(type) : fullConeType(type);
     }
 
     if (type == Type::unreachable) {
```
That applies exactness on creation. I think it should be enough here?

Reply:
It looks like `fullConeType` is also used directly in several places. WDYT about having this check in `fullConeType` so it's harder to inadvertently skip?

Reply:
Ah, good point. Yeah, that seems safest.